app/code/Bulbulatory/Recommendations/view/adminhtml/ui_component/bulbulatory_recommendations_table_listing.xml

```diff
@@ -0,0 +1,111 @@
+<listing xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:module:Magento_Ui:etc/ui_configuration.xsd">
+    <argument name="data" xsi:type="array">
+        <item name="js_config" xsi:type="array">
+            <item name="provider" xsi:type="string">bulbulatory_recommendations_table_listing.bulbulatory_recommendations_table_listing_data_source</item>
+            <item name="deps" xsi:type="string">bulbulatory_recommendations_table_listing.bulbulatory_recommendations_table_listing_data_source</item>
+        </item>
+        <item name="spinner" xsi:type="string">spinner_columns</item>
+    </argument>
+    <dataSource name="nameOfDataSource">
+        <argument name="dataProvider" xsi:type="configurableObject">
+            <argument name="class" xsi:type="string">Magento\Framework\View\Element\UiComponent\DataProvider\DataProvider</argument>
+            <argument name="name" xsi:type="string">bulbulatory_recommendations_table_listing_data_source</argument>
+            <argument name="primaryFieldName" xsi:type="string">id</argument>
+            <argument name="requestFieldName" xsi:type="string">id</argument>
+            <argument name="data" xsi:type="array">
+                <item name="config" xsi:type="array">
+                    <item name="component" xsi:type="string">Magento_Ui/js/grid/provider</item>
+                    <item name="update_url" xsi:type="url" path="mui/index/render"/>
+                    <item name="storageConfig" xsi:type="array">
+                        <item name="indexField" xsi:type="string">id</item>
+                    </item>
+                </item>
+            </argument>
+        </argument>
+    </dataSource>
+    <listingToolbar name="listing_top">
+        <paging name="listing_paging"/>
+    </listingToolbar>
+    <columns name="spinner_columns">
+        <selectionsColumn name="ids">
+            <argument name="data" xsi:type="array">
+                <item name="config" xsi:type="array">
+                    <item name="resizeEnabled" xsi:type="boolean">false</item>
+                    <item name="resizeDefaultWidth" xsi:type="string">55</item>
+                    <item name="indexField" xsi:type="string">id</item>
+                </item>
+            </argument>
+        </selectionsColumn>
+        <column name="id">
+            <argument name="data" xsi:type="array">
+                <item name="config" xsi:type="array">
+                    <item name="filter" xsi:type="string">textRange</item>
+                    <item name="sorting" xsi:type="string">asc</item>
+                    <item name="label" xsi:type="string" translate="true">ID</item>
+                </item>
+            </argument>
+        </column>
+        <column name="customer_id" class="Bulbulatory\Recommendations\Ui\Component\Listing\Column\Action">

```

Comment:
The name of the column is `customer_id` but the data inside it is its email address. It's not good to mix things like that. If the column shows an email, it should be named appropriately. It would also require a change in https://github.com/quliyevrustam/Academy/pull/7/files#diff-ca6eefaec4006a8a8fdd2656382ff5c7R46 - if you name the column eg. `customer_email`, you should also set a key `customer_email` in `$item` array.

Reply:
@TM18

> The name of the column is customer_id but the data inside it is its email address. It's not good to mix things like that. If the column shows an email, it should be named appropriately. It would also require a change in https://github.com/quliyevrustam/Academy/pull/7/files#diff-ca6eefaec4006a8a8fdd2656382ff5c7R46 - if you name the column eg. customer_email, you should also set a key customer_email in $item array.

Fixed!